src/main/java/com/example/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,64 @@
+package com.example.controller;
+
+import com.example.dto.cart.CreateCartItemRequestDto;
+import com.example.dto.cart.ShoppingCartDto;
+import com.example.dto.cart.UpdateCartItemRequestDto;
+import com.example.model.User;
+import com.example.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing cart")
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @Operation(summary = "Get cart", description = "Get cart of user")
+    @GetMapping
```

Comment:
Add `@PreAuthorize` where is needed

Reply:
I don't realize why it need add @PreAuthorize User. This app don't have controllers methods without Admin access.